### PR TITLE
[flutter_tools] reland: move process resolution into tool.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Need to run a global g3 run to catch the missed tests

Reland of https://github.com/flutter/flutter/pull/75350

### CODE REVIEW

**Comment by jonahwilliams** [comment]
FYI @christopherfujino

**Comment by jonahwilliams** [comment]
Same as last time except I added back the default injection. Otherwise needs LGTM for g3 testing.

**Review by zanderso — APPROVED** [review]

**Comment by jonahwilliams** [comment]
In order to make this landable without updating a bunch of mocks/tests, I'm going to make the tool process manager implement the existing package:process manager and then add special handling to the injector. Then I can clean up g3 in a non-blocking manner

File: packages/flutter_tools/lib/src/base/context.dart [hunk 1]
```diff
@@ -116,7 +119,13 @@ class AppContext {
   /// Gets the value associated with the specified [type], or `null` if no
   /// such value has been associated.
   T get<T>() {
-    dynamic value = _generateIfNecessary(T, _overrides);
+    // Convert lookups of old process manager into new process manager.
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/base/context.dart:122]: Does this need a TODO to track removing after the internal usage is sorted out?

File: packages/flutter_tools/lib/src/base/context.dart [hunk 2]
```diff
@@ -116,7 +119,13 @@ class AppContext {
...
+    dynamic value;
+    if (T == process.ProcessManager) {
+       value = _generateIfNecessary(ProcessManager, _overrides);
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/base/context.dart:125]: nit: check indentation.

File: packages/flutter_tools/lib/src/base/context.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/base/context.dart:122]: depends on if it works or not :)

File: packages/flutter_tools/lib/src/base/context.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/base/context.dart]: fixed

**Comment by jonahwilliams** [comment]
Hmm, this still won't work

**Comment by jonahwilliams** [comment]
The implement both APIs doesn't really work, I'm looking into the size of the diff for an atomic change

**Comment by jonahwilliams** [comment]
It will not be feasible to make this change. i'm investigating other avenues to get the behavior we'd want.

**Pull request closed without merging**